src/utils/thirdPartyConfig.ts

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable sort-keys */
+
+import * as t from "io-ts";
+
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+
+import { fallback, jsonFromString } from "../types/commons";
+
+export type AuthenticationConfig = t.TypeOf<typeof AuthenticationConfig>;
+export const AuthenticationConfig = t.interface({
+  // Right now we only handle API_KEY
+  // In future we will also handle BEARER and TOKEN authentications
+  type: t.literal("API_KEY"),
+  parameterName: NonEmptyString,
+  required: t.boolean
+});
+
+export type ThirdPartyConfig = t.TypeOf<typeof ThirdPartyConfig>;
+export const ThirdPartyConfig = t.intersection([
+  t.interface({
+    serviceId: NonEmptyString,
+    baseUrl: NonEmptyString,
+    kind: NonEmptyString,
+    jsonSchema: NonEmptyString,
+    detailsAuthentication: AuthenticationConfig
+  }),
+  t.partial({
+    attachmentsAuthentication: AuthenticationConfig
```

Comment:
I'm wondering about this: What if a production service needs attachmentsConfiguration and we miss it ? Can we avoid misconfiguration introducing an attribute that works as pivot for disjoint?

Reply:
I'm wondering the value of defining this pivot: if we miss attachmentsConfiguration because we don't know it will be used, this field does not save use. On the other hand, if we know that attachmentsAuthentication, we just add it.

Reply:
I'm wondering if we really want to have different authentication method for message content and attachment.
Maybe we could force a single authentication config used for both.